core/src/error/suggest.rs

```diff
@@ -0,0 +1,49 @@
+//! Suggest existing symbols that are similar to what the user typed in case of various "symbol not
+//! found" errors, such as [super::EvalError::UnboundIdentifier] or
+//! [super::EvalError::FieldMissing].
+//!
+//! The current implementation uses the normalized Damereau-Levenshtein edit distance to find the
+//! closest match.
+
+use strsim::normalized_damerau_levenshtein;
+
+/// The minimum similarity between the user's input and an existing symbol for the symbol to be
+/// considered a serious candidate. The current threshold is rather low, because short words with
+/// edit distance 1 such as `bar` and `bare` might have a similarity which isn't that high.
+pub const MIN_SIMILARITY: f64 = 0.60;
```

Comment:
If the right threshold changes depending on the length, is it worth having a different threshold for short words?

Reply:
I'm not sure. It would be worth it if this threshold was giving obviously bad suggestion for long words, which doesn't seem to the case (but it could be, I've just toyed with a few examples, to be honest). All in all it's just a threshold, so we still pick the most similar world possible. I would say let's see and if this does happen, maybe we will switch to several threshold, or just use the goold old edit distance instead of the normalized one.